nx/lib/nx/defn/expr.ex

```diff
@@ -166,6 +204,7 @@ defmodule Nx.Defn.Expr do
       expr
       |> Nx.as_type(type)
       |> Nx.broadcast(shape, names: names)
+      |> Nx.devectorize(keep_names: false)
```

Comment:
@josevalim I found a problem here.
vectorized broadcasting has an "axis completion" algorithm for broadcasting scalars against vectorized tensors.
The easiest way to deal with this is to keep `broadcast_vectors` above instead of `reshape_vectors`.

As the code is commited, it doesn't deal with branches with different vectorized axes, as per the single failing test we have.

Otherwise, I think we would need to replicate most of the logic for broadcast_vectors and some of the logic for broadcast.

WDYT?